app/src/main/java/com/abhijitvalluri/android/fitnotifications/services/NLService.java

```diff
@@ -331,65 +294,28 @@ public void onNotificationPosted(final StatusBarNotification sbn) {
         stackBuilder.addParentStack(SettingsActivity.class);
         // Adds the Intent that starts the Activity to the top of the stack
         stackBuilder.addNextIntent(settingsIntent);
-        PendingIntent settingsPendingIntent =
-                stackBuilder.getPendingIntent(
-                        0,
-                        PendingIntent.FLAG_UPDATE_CURRENT
-                );
+        PendingIntent settingsPendingIntent = stackBuilder.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT);
         builder.setContentIntent(settingsPendingIntent).setAutoCancel(true);
 
-        StringBuilder notifStrB = new StringBuilder(sb.toString().trim().replaceAll("\\s+", " "));
+        String notificationString = sb.toString().trim().replaceAll("\\s+", " ");
 
-        if (mNotificationStringMap.containsKey(appPackageName)) {
-            String prevNotificationString = mNotificationStringMap.get(appPackageName);
-            if (prevNotificationString.equals(notifStrB.toString())) {
-                mNotificationStringMap.remove(appPackageName);
-                return;
-            }
+        String prevNotificationString = mNotificationStringMap.put(appPackageName, notificationString);
+        if (notificationString.equals(prevNotificationString)) {
+            // do not repeat the same notification
```

Comment:
But then outside of that special scenario you will be blocking legitimate second notifications.

I'll revert the change so that at least only every 2nd notification is blocked. But I believe some more checks for that specific scenario are needed here.